### ISSUE TITLE
Applying ZCA Whithening on cifar-10

### ISSUE THREAD

**Issue by zeno40** [OpenANN/OpenANN#175]
Hi, first of all thanks for your great project! I'm currently try to apply ZCA Whithening to the
cifar-10 dataset but I'm not sure I do it the right way:

`Eigen::MatrixXd trainingInput;`
`trainingInput.resize(50000, 3072);`

`// .... load cifar-10 in matrix trainingInput`
`// ... Scale data to [-1, 1]`

`ZCAWhitening whithening;`
`trainingInput = whithening.fit(trainingInput).transform(trainingInput);`

is this correct?

thanks,
Filip

**Comment by AlexanderFabisch** [comment]
Yes, see [here](https://github.com/OpenANN/OpenANN/blob/master/test/ZCATestCase.cpp#L25) for an example.

**Comment by zeno40** [comment]
Thanks a lot!
Apply the ZCAWhithening transformation is a very lenghty process for the cifar-10 training set. Is it possible to save  the transformed matrix back in unsigned char(byte) variables so that I don't have to apply the whithening transformation each time?

Filip

**Comment by zeno40** [comment]
In the  cifar-10 loader you provided you divide by 128 and substract by 1. Is this correct? Don't we have to divide by 127.5 and substract by 1 to get in the range of [-1, 1]?

(255/128) - 1 = 0,9921875
(255/127.5) -1 = 1

thanks,
Filip

**Comment by AlexanderFabisch** [comment]
You are right. You can open a pull request if you want. :)
